src/controllers/accommodation.js

```diff
@@ -258,6 +262,99 @@ class AccommodationController {
       ErrorResponse.internalServerError(res, error.message);
     }
   };
+
+  static addFeedback = async (req, res) => {
+    try {
+      const findAccommodation = await Accommodation.findOne({
+        where: { id: req.params.id },
+      });
+      if (!findAccommodation) {
+        return ErrorResponse.notFoundError(res, 'Accommodation not found');
+      }
+
+      const trips = await req.user.getTripRequests({
+        where: { accommodationId: findAccommodation.id },
+      });
+      const hasStayed = trips.find((trip) => isBeforeToday(trip.departureDate));
+
+      if (!hasStayed) {
+        return ErrorResponse.semanticError(
+          res,
+          `You need to have stayed at least one day in an accommodation to give it a feedback`
+        );
+      }
+
+      const provideFeedback = await findAccommodation.addUser(req.user, {
+        through: { feedback: req.body.feedback },
+      });
+      return res.status(201).json({
+        status: '201',
+        message: 'Feedback added successfully',
+        payload: provideFeedback,
+      });
+    } catch (error) {
+      ErrorResponse.internalServerError(res, error.message);
+    }
+  };
+
+  static getFeedbacks = async (req, res) => {
+    try {
+      const feedback = await UserAccommodation.findAll({
+        where: { accommodationId: req.params.id },
+      });
+      return res.status(200).json({
+        status: '200',
+        message: 'These are all feedbacks of this accommodation',
+        payload: feedback,
+      });
+    } catch (error) {
+      ErrorResponse.internalServerError(res, error.message);
+    }
+  };
+
+  static updateFeedback = async (req, res) => {
+    try {
+      const findAccommodation = await Accommodation.findOne({
+        where: { id: req.params.id },
+      });
+      if (!findAccommodation) {
+        return ErrorResponse.notFoundError(res, 'Accommodation not found');
+      }
+
+      const updatedFeedback = await findAccommodation.addUser(req.user, {
+        through: { feedback: req.body.feedback },
+      });
+      return res.status(200).json({
+        status: '200',
+        message: 'Feedback updated successfully',
+        payload: updatedFeedback,
+      });
+    } catch (error) {
+      ErrorResponse.internalServerError(res, error.message);
+    }
+  };
+
+  static deleteFeedback = async (req, res) => {
+    try {
+      const findAccommodation = await Accommodation.findOne({
+        where: { id: req.params.id },
+      });
+      if (!findAccommodation) {
+        return ErrorResponse.notFoundError(res, 'Accommodation not found');
+      }
+
+      const deletedFeedback = await UserAccommodation.destroy({
+        where: { accommodationId: req.params.id },
+      });
```

Comment:
Can you add the userId parameter as well, to delete only the feedback by that user instead of all the feedbacks on the mentioned accommodations

Reply:
Yeah, let me fix this too